src/core/components/layout/components/columns/styles.ts

```diff
@@ -16,33 +17,21 @@ export const columns = css`
 	}
 `;
 
-const collapseBelowSpacing = css`
+const collapseBelowSpacing = (spaceY?: ColumnsSpaceY) => css`
 	display: block;
 	margin-right: 0;
 	& > * {
 		margin-right: 0;
-		margin-bottom: ${space[5]}px;
+		${spaceY ? `margin-bottom: ${space[spaceY]}px;` : ''}
 	}
 `;
 
-export const collapseBelowTabletColumns = css`
-	${until.tablet} {
-		${collapseBelowSpacing}
-	}
-`;
-export const collapseBelowDesktopColumns = css`
-	${until.desktop} {
-		${collapseBelowSpacing}
-	}
-`;
-export const collapseBelowLeftColColumns = css`
-	${until.leftCol} {
-		${collapseBelowSpacing}
-	}
-`;
-export const collapseBelowWideColumns = css`
-	${until.wide} {
-		${collapseBelowSpacing}
+export const collapseBelowColumns = (
+	breakpoint: CollapseBreakpoint,
+	spaceY?: ColumnsSpaceY,
+) => css`
+	${until[breakpoint]} {
+		${collapseBelowSpacing(spaceY)}
```

Comment:
I like to avoid using dynamic styles if possible. Dynamic styles can't be determined at build time and therefore will always require a runtime to be shipped to the client. In the future, I'd love for us to move away from Emotion to a solution like [Treat](https://seek-oss.github.io/treat/) or [Linaria](https://github.com/callstack/linaria) that doesn't require a runtime, and doesn't support dynamic styles.

To achieve this, we can use a pattern similar to that in the [Stack styles](https://github.com/guardian/source/blob/4c2c1baaa9e2f34576d1f7d206baf6df7a26a826/src/core/components/layout/components/stack/styles.ts#L17-L37): we expose a map (or just a plain object) with breakpoints for keys and calls to a function that generates the CSS. This keeps duplication low, but means the styles can (theoretically at least) be determined at build time.

```ts
const collapseBelowColumnsCSS = (breakpoint: Breakpoint) => css`
  ${until[breakpoint]} {
    display: block;
    margin-right: 0;
    & > * {
      margin-right: 0;
    }
  }
` ;

export const collapseBelowColumns = {
  tablet: collapseBelowColumnsCSS('tablet'),
  //...
};
```

What about spacing? We can almost lift this straight out the Stack styles

```ts
const collapseBelowSpaceYCSS (spaceY: ColumnsSpaceY) => css`
  & > * {
    margin-bottom: ${space[spaceY]}px;
  }
`;

export const collapseBelowSpaceY: {
	1: SerializedStyles;
	2: SerializedStyles;
} = {
	1: collapseBelowSpaceYCSS(1),
	2: collapseBelowSpaceYCSS(2),
    //...
};
```

But won't this apply the spaceY margin at every breakpoint? Uhhh... yeah, good point! 😅  Maybe we can correct for this also applying a negative bottom margin to the column component?

```ts
const collapseBelowSpaceYCSS (spaceY: ColumnsSpaceY) => css`
  margin-bottom: ${-space[spaceY]}px;
  & > * {
    margin-bottom: ${space[spaceY]}px;
  }
`;
```

Let me know what you think 😄

Reply:
That makes sense, and thanks for the detailed comment. 👍 

Hopefully https://github.com/guardian/source/pull/776/commits/a3fbea5aa5eb5a682e63536363d8842b344c5934 addresses this.